oak-jcr/src/main/java/org/apache/jackrabbit/oak/jcr/session/SessionImpl.java

```diff
@@ -773,6 +785,122 @@ public UserManager getUserManager() throws RepositoryException {
         return sessionContext.getUserManager();
     }
 
+    //--------------------------------------------------< HttpBinaryProvider >---
+
+    @Nullable
+    @Override
+    public BinaryHttpUpload initiateHttpUpload(String path, long maxSize, int maxParts) throws RepositoryException {
+        if (! hasPermission(path, Session.ACTION_SET_PROPERTY)) {
+            throw new AccessDeniedException(String.format("No permission to add binary property at path %s", path));
+        }
+        return sd.safePerformNullable(new ReadOperation<BinaryHttpUpload>("initiateHttpUpload") {
```

Comment:
@Oak experts: Using `ReadOperation` here (and in 2 new methods below) which only does an is-session-alive check (which we want). This is not a write as it does not change the repository (NodeStore) in any way, and it's also not technically a read from the repository, but it seems that `ReadOperation` does not particularly care.